ui/pages/Project/components/FamilyTable/CollapsableLayout.jsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+class CollapsableLayout extends React.PureComponent {
+
+  static propTypes = {
+    layoutComponent: PropTypes.elementType.isRequired,
+    detailFields: PropTypes.arrayOf(PropTypes.object).isRequired,
+    noDetailFields: PropTypes.arrayOf(PropTypes.object),
+  }
+
+  state = { showDetails: false }
+
+  toggle = () => {
+    this.setState(prevState => ({ showDetails: !prevState.showDetails }))
```

Comment:
Using an object parameter for the `setState` function should be sufficient.

Reply:
I don't understand what that means. This is adapted from a copy-paste

Reply:
Using a function for the setState parameter allows access to the `latest` state value after previous calls to the `setState`. We don't have multiple calls to the `setState` before re-rendering. So it is safe and neater to use an object parameter like `this.setSate({ showDetails: !prevState.showDetails })`.

Reply:
thats a violation of our linter and less stable than the function approach used here